Battle.js

```diff
@@ -0,0 +1,124 @@
+/*TODO: 
+Main battle screen, should display 2 knights with health bars and a 
+button for each different ability*/
+
+import { StyleSheet, Text, View, Button, Image, Dimensions, Pressable } from 'react-native';
+import React, { useState, useRef, useEffect } from 'react';
+import { HealthBar } from "./HealthBar.js"
+import { score } from "./Dodge.js"
+import { useFocusEffect } from '@react-navigation/native';
+import { hookstate, useHookstate, createState } from '@hookstate/core';
+// .value: Gets the value of the state
+// .set(): Sets the value of the state
+
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+//const [health, setHealth] = useState(100);
+const health2 = createState(100);
+
+export default function Battle({ navigation, route }) {
```

Comment:
This could very naturally be a class, not just a function.

Then you can export the class (which will contain the healthbar), rather than exporting them each separately.

[This is more of the "once you're using classes, everything is a class" mentality, but I *do* think it works well for things that correspond naturally to a single HTML component, which contains some data about that component (e.g. the current health) and some more basic things (like how to render that component).